bin/check_sm_processing_job.py

```diff
@@ -0,0 +1,138 @@
+#  Copyright 2024 Amazon.com, Inc. or its affiliates.
+
+import argparse
+from datetime import datetime
+from typing import Dict, List, Tuple, Union
+
+import boto3
+from boto3.resources.base import ServiceResource
+
+
+def get_processing_job(sagemaker_client: ServiceResource, job_name: str) -> Tuple[str, str, str, str]:
+    """
+    Get the latest status of a SageMaker processing job.
+
+    :param sagemaker_client: The SageMaker client.
+    :param job_name: The name of the SageMaker processing job.
+
+    :returns: A tuple containing the job status, duration, creation time, and CloudWatch URL.
+    """
+    try:
+        response = sagemaker_client.describe_processing_job(ProcessingJobName=job_name)
+        status = response["ProcessingJobStatus"]
+
+        creation_time = response["CreationTime"]
+
+        if "ProcessingStartTime" not in response or "ProcessingEndTime" not in response:
+            time_now = datetime.now().astimezone()
+            duration = duration_in_human_readable(creation_time, time_now)
+        else:
+            processing_start_time = response["ProcessingStartTime"]
+            processing_end_time = response["ProcessingEndTime"]
+            duration = duration_in_human_readable(processing_start_time, processing_end_time)
+
+        log_group = response.get("ProcessingJobArn").split(":processing-job/")[1]
+        log_stream = response.get("ProcessingJobArn").split(":processing-job/")[1]
+        region_name = sagemaker_client.meta.region_name
+        cw_endpoint = f"https://{region_name}.console.aws.amazon.com/cloudwatch/home"
+        cloudwatch_url = (
+            f"{cw_endpoint}?region={region_name}#logsV2:log-groups/log-group/{log_group}/log-events/{log_stream}"
+        )
+
+        return status, duration, creation_time.strftime("%Y-%m-%d %H:%M:%S %Z"), cloudwatch_url
+    except sagemaker_client.exceptions.ResourceNotFound:
+        return "Processing job not found."
```

Comment:
This return does not match the defined return type.  Would it make sense to let this bubble up the same as other exceptions and handle it there?